p2panda-rs/src/document/document_id.rs

```diff
@@ -23,13 +23,13 @@ use crate::{Human, Validate};
 ///                         \
 ///                          \__ [UPDATE] (Hash: "eff..")
 /// ```
-#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
```

Comment:
`Serialize` and `Deserlialize` traits are gone in this PR but they are needed for gql scalar conversions in `aquadoggo`. Any reason they should be gone? @adzialocha @cafca ?

Reply:
I've removed all serde implementations which are not required directly, of course without looking at aquadoggo though .. if we need them we can bring them back but bear in mind to not just derive `Deserialize` as it would just throw anything unchecked into the underlying string type.

That being said, is it really required in aquadoggo? I understand that the scalar trait actually doesn't use serde, the derives there are unused and misleading, should probably be removed.

I'm not sure all of this is correct, haven't used it yet, I just have a feeling 😅

Reply:
My understanding is that Piet used these for replication but `async_graphql` actually has its own system for scalar conversions (`ScalarType`) that _can_ use serde but doesn't have to.

Reply:
Ok, thanks for the comments, I'm unfamiliar with this particular corner of the doggo. I'll check if the traits are actually needed and remove or re-look at it.